### PR TITLE
DDCNL-279 - Remove SA Ambiguous Old Journey Code.

### PR DESCRIPTION
**Pull request by RobertLaverick**
This reverts commit 193f62a7ed28d788ad660e7bd7111ee1c896755a.

Intent is to remove the old self-assessment journey option.

### CODE REVIEW

File: app/controllers/AmbiguousJourneyController.scala [hunk 1]
```diff
@@ -198,45 +187,45 @@ class AmbiguousJourneyController @Inject()(
 
   def handleAmbiguousJourneyLandingPages(page: String): Action[AnyContent] =
     (authenticate andThen enforceAmbiguousUserAction) { implicit request =>
+      val continueUrl = controllers.routes.HomeController.index().url
       request.saUserType match {
-        case NotEnrolledSelfAssessmentUser(saUtr)      => handleAmbiguousJourney(page, saUtr)
-        case WrongCredentialsSelfAssessmentUser(saUtr) => handleAmbiguousJourney(page, saUtr)
-        case _                                         => Redirect(routes.HomeController.index())
+        case NotEnrolledSelfAssessmentUser(saUtr) =>
```
Comment (oc-ste) [inline app/controllers/AmbiguousJourneyController.scala]: this is old SA code

File: app/views/ambiguousjourney/filedReturnOnlineChoice.scala.html [hunk 2]
```diff
@@ -26,7 +26,7 @@
 
 @main(title = messages("label.have_you_ever_filed_your_tax_return_online"),
     pageName = Some(messages("label.your_personal_tax_account")),
-    backlinkUrl = Some(controllers.routes.AmbiguousJourneyController.landingPage().url)
+    backlinkUrl = Some(controllers.routes.SelfAssessmentController.handleSelfAssessment().url)
```
Comment (oc-ste) [inline app/views/ambiguousjourney/filedReturnOnlineChoice.scala.html]: is this view still used?

File: app/views/selfAssessmentNotShown.scala.html [hunk 3]
```diff
@@ -19,7 +19,7 @@
 @import uk.gov.hmrc.renderer.TemplateRenderer
 @import util.LocalPartialRetriever
 
-@(saUtr: Option[SaUtr])(implicit request: UserRequest[_], configDecorator: ConfigDecorator, partialRetriever: LocalPartialRetriever, templateRenderer: TemplateRenderer, messages: play.api.i18n.Messages)
+@(utr: SaUtr)(implicit request: UserRequest[_], configDecorator: ConfigDecorator, partialRetriever: LocalPartialRetriever, templateRenderer: TemplateRenderer, messages: play.api.i18n.Messages)
```
Comment (oc-ste) [inline app/views/selfAssessmentNotShown.scala.html]: unused view?

File: conf/prod.routes [hunk 4]
```diff
@@ -32,7 +32,6 @@ GET         /personal-account/self-assessment/sign-in-again
 GET         /personal-account/self-assessment/need-to-reset-password                   @controllers.SaWrongCredentialsController.needToResetPassword
 GET         /personal-account/self-assessment/find-your-user-id                        @controllers.SaWrongCredentialsController.findYourUserId
 
-GET         /personal-account/self-assessment/ambiguous-sa-user                        @controllers.AmbiguousJourneyController.landingPage
 GET         /personal-account/self-assessment/sa-filed-online                          @controllers.AmbiguousJourneyController.filedReturnOnlineChoice
```
Comment (oc-ste) [inline conf/prod.routes]: can remove all these below

File: test/controllers/AmbiguousJourneyControllerSpec.scala [hunk 5]
```diff
@@ -66,808 +68,803 @@ class AmbiguousJourneyControllerSpec extends BaseSpec with MockitoSugar {
...
-          when(mockConfigDecorator.saAmbigSimplifiedJourneyEnabled) thenReturn false
-          when(mockConfigDecorator.saAmbigSkipUTRLetterEnabled) thenReturn false
-
-          val page = "pin-expired"
-          val result = controller.handleAmbiguousJourneyLandingPages(page)(FakeRequest())
-
-          status(result) shouldBe 200
-        }
-      }
+  val ambiguousUser = NotEnrolledSelfAssessmentUser(SaUtr("1111111111"))
```
Comment (oc-ste) [inline test/controllers/AmbiguousJourneyControllerSpec.scala]: reinstated an unused test?

File: test/controllers/auth/EnforceAmbiguousUserActionSpec.scala [hunk 6]
```diff
@@ -53,27 +52,51 @@ class EnforceAmbiguousUserActionSpec extends FreeSpec with MustMatchers with Gui
...
-
-            val userRequest = buildUserRequest(saUser = selfAssessmentType, request = FakeRequest())
-
-            val result = harness()(userRequest)
-            status(result) mustBe OK
-          }
-        }
+      "return the request it was passed" in {
+        val userRequest =
+          UserRequest(
```
Comment (oc-ste) [inline test/controllers/auth/EnforceAmbiguousUserActionSpec.scala]: Clean up creation by using buildUserRequest

File: test/controllers/auth/EnforceAmbiguousUserActionSpec.scala [hunk 7]
```diff
@@ -53,27 +52,51 @@ class EnforceAmbiguousUserActionSpec extends FreeSpec with MustMatchers with Gui
...
-        val userRequest = buildUserRequest(saUser = NonFilerSelfAssessmentUser, request = FakeRequest())
+      "when a user is not ambiguous" - {
 
-        val result = harness()(userRequest)
-        status(result) mustBe SEE_OTHER
-        redirectLocation(result).get must endWith("/personal-account")
+        "redirect to the landing page" in {
+          val userRequest =
+            UserRequest(
+              None,
```
Comment (oc-ste) [inline test/controllers/auth/EnforceAmbiguousUserActionSpec.scala]: Clean up creation by using buildUserRequest

File: app/controllers/AmbiguousJourneyController.scala (hunk 1 above)
Comment (RobertLaverick) [reply app/controllers/AmbiguousJourneyController.scala]: Note - bin controller

File: app/controllers/auth/EnforceAmbiguousUserAction.scala [hunk 8]
```diff
@@ -28,8 +28,8 @@ class EnforceAmbiguousUserAction
 
   override protected def refine[A](request: UserRequest[A]): Future[Either[Result, UserRequest[A]]] =
     request.saUserType match {
-      case _: NotEnrolledSelfAssessmentUser | _: WrongCredentialsSelfAssessmentUser => Future.successful(Right(request))
-      case _                                                                        => Future.successful(Left(Redirect(routes.HomeController.index())))
+      case _: NotEnrolledSelfAssessmentUser => Future.successful(Right(request))
```
Comment (oc-ste) [inline app/controllers/auth/EnforceAmbiguousUserAction.scala]: need this OR in | _: WrongCredentialsSelfAssessmentUser

**Comment by oc-ste** [comment]
@RobertLaverick remove unused URLs from configDecorator

**Pull request merged**